### PR TITLE
New macOS SDK

### PR DESCRIPTION
**Pull request by isuruf**
cc @conda-forge/core, @mingwandroid

### CODE REVIEW

File: src/maintainer/knowledge_base.rst [hunk 1]
```diff
@@ -748,3 +748,29 @@ Following implies that ``python`` is a runtime dependency and a Python matrix fo
 
 You need to rerender the feedstock after this change.
 
+Requiring newer macOS SDKs
+==========================
+
+conda-forge uses macOS SDK 10.9 to build software so that they can be deployed to
+all macOS versions newer than 10.9. Sometimes, some packages require a newer SDK
+to build with. While the default version 10.9 can be overridden using the following
+changes to the recipe, it should be done as a last resort.
```
Comment (jakirkham) [inline src/maintainer/knowledge_base.rst]: Should we add something like, "Please consult with core if this is something you think you need."?

Comment (isuruf) [reply src/maintainer/knowledge_base.rst]: Sure. Please go ahead and push to this branch.

**Comment by isuruf** [comment]
@mariusvniekerk, @CJ-Wright, does cflib have data about which packages used which MACOSX_DEPLOYMENT_TARGET ?  We need to patch them to add the `run_constrained`.

**Comment by jakirkham** [comment]
Could we hot fix them?

**Comment by isuruf** [comment]
Yes, we can hot fix them and we also need send PRs to those feedstocks.

**Comment by jakirkham** [comment]
That's a pretty large task. Could we scope this out somewhere? Perhaps in a new issue?

**Comment by jakirkham** [comment]
I'm guessing we will want to do something similar for Linux too.

**Comment by isuruf** [comment]
There aren't many feedstocks using SDKs greater than 10.9. Are there?

**Comment by jakirkham** [comment]
Won’t we want to add this to the ones that are too?

**Comment by jakirkham** [comment]
It’s worth noting some of the things that will need this are repackaged binaries. Not totally sure how we identify those.

**Comment by isuruf** [comment]
conda installed from miniconda and anaconda will not run on OSX<10.9, so that's not a huge issue.

**Comment by mingwandroid** [comment]
> It’s worth noting some of the things that will need this are repackaged binaries. Not totally sure how we identify those.

We would need to look at the used symbols from libSystem.dylib for this, should be possible. Small exes that use only old symbols (despite being compiled with newer SDKs) would get marked with an older SDK, but that should be fine (I think/hope) .. failing that, manual testing on different macOS would be needed I suppose.

**Pull request merged**